src/index.c

```diff
@@ -781,7 +781,9 @@ IndexIterator *NewIntersecIterator(IndexIterator **its_, size_t num, DocTable *d
   ctx->num = num;
 
   // Sort children iterators from low count to high count which reduces the number of iterations.
-  qsort(ctx->its, ctx->num, sizeof(*ctx->its), (CompareFunc)cmpIter);
+  if (!ctx->inOrder) {
```

Comment:
@ashtul would it be better to save 2 arrays, one is sorted to be optimized and the other is for check slop and inorder?

Reply:
@MeirShpilraien I considered it, but we check for `INORDER` in the same order in which we picked the results from the sub-iterators (`RSIndexResult_HasOffsets(r->children[i])`).
Therefore, the change suggested requires sorting results according to the original order for each matching intersect. So while it is doable, it seems like a complexity that (1) might cost more than whatever was optimized and (2) would be hard and unintuitive to maintains. Removing the optimization when INORDER is used seems like an adequate and elegant solution.

WDYT?